.github/workflows/test.yml

```diff
@@ -174,15 +174,15 @@ jobs:
           key: ${{ runner.os }}-hf-model-${{ env.model_key }}
 
       - name: "Download HF models"
-        if: ( steps.changed-src-files.outputs.any_changed == 'true' && steps.cache_restore.outputs.cache-hit != 'true' )
+        if: ( steps.changed-src-files.outputs.any_changed == 'true' && steps.cache_restore.outputs.cache-hit == 'true' )
```

Comment:
This means you will have both revisions:
- 4 layer model from the cache
- 3 layer model downloaded with revision

Making all the tests work. If you bypass the cache, I reckon some tests will break like in PR #499

Reply:
Once the 3 layer model is verified, I don't think we need the 4 layer anymore, so we should only be using one revision of the model at all places

Reply:
I think it needs more code/test changes to make sure we don't use the latest/main revision of the tiny granite model anymore/anywhere.

Kicking of a test run here: #502

Reply:
#503